llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -1427,6 +1432,8 @@ bool SPIRVInstructionSelector::selectIntrinsic(Register ResVReg,
           .addUse(I.getOperand(2).getReg())
           .addUse(I.getOperand(3).getReg());
     break;
+  case Intrinsic::dx_thread_id:
+    return selectDXThreadId(ResVReg, ResType, I);
```

Comment:
This feels weird to me. Using an intrinsic from another target isn't something I've ever seen done before in LLVM.

Reply:
Thanks, I was surprised that it worked but didn't know whether that indicated it was by design or accident. I added an analogous SPIR-V intrinsic.